int8 check
float64 x
float64 y
float64 dist
float64 angle
